airflow/models/dag.py

```diff
@@ -348,6 +348,12 @@ def __init__(
         self.partial = False
         self.on_success_callback = on_success_callback
         self.on_failure_callback = on_failure_callback
+
+        # To keep it in parity with Serialized DAGs
+        # and identify if DAG has on_*_callback without actually storing them in Serialized JSON
+        self.has_on_success_callback = bool(self.on_success_callback is not None)
+        self.has_on_failure_callback = bool(self.on_failure_callback is not None)
```

Comment:
```suggestion
        self.has_on_success_callback = self.on_success_callback is not None
        self.has_on_failure_callback = self.on_failure_callback is not None
```

No need for bool I don't think

Reply:
Oh yea not needed now.

Previously I had:

```python
        self.has_on_success_callback = bool(self.on_success_callback)
        self.has_on_failure_callback = bool(self.on_failure_callback)
```

before I changed it couple of times :D